marklogic-data-hub-central/ui/src/components/explore/graph-vis-explore/graph-vis-explore.tsx

```diff
@@ -532,24 +525,30 @@ const GraphVisExplore: React.FC<Props> = (props) => {
     };
     try {
       let response: any;
+      let checkThreshold = false;
       if (!payloadData.expandAll) {
         response = await expandGroupNode(payload, 3);
       } else {
         response = await expandGroupNode(payload);
+        checkThreshold = true;
       }
 
       if (response.status === 200) {
-        let expandedNodeInfo = expandedNodeData;
-        let expandId = payloadData["nodeInfo"] ? payloadData.nodeInfo["expandId"] : clickedNode["expandId"];
-        expandedNodeInfo[expandId] = {
-          nodes: response.data.nodes,
-          edges: response.data.edges,
-          removedNode: payload.data["parentIRI"]
-        };
-        setExpandedNodeData(expandedNodeInfo);
-        network.body.data.nodes.remove(payload.data["parentIRI"]);
-        await updateGroupAndLeafNodesDataset(response.data.nodes);
-        await updateGraphDataWith(response.data.nodes, response.data.edges);
+        if (checkThreshold && response.data.total > 1000) {
+          handleTableViewRecords("exceededThreshold");
+        } else {
+          let expandedNodeInfo = expandedNodeData;
+          let expandId = payloadData["nodeInfo"] ? payloadData.nodeInfo["expandId"] : clickedNode["expandId"];
+          expandedNodeInfo[expandId] = {
+            nodes: response.data.nodes,
+            edges: response.data.edges,
+            removedNode: payload.data["parentIRI"]
+          };
+          setExpandedNodeData(expandedNodeInfo);
+          network.body.data.nodes.remove(payload.data["parentIRI"]);
+          await updateGroupAndLeafNodesDataset(response.data.nodes);
+          await updateGraphDataWith(response.data.nodes, response.data.edges);
```

Comment:
If the that calls don't have dependency in the result the performance can be improved doing something like this ```await Promise.all([updateGroupAndLeafNodesDataset(response.data.nodes),  updateGraphDataWith(response.data.nodes, response.data.edges)])```

Reply:
That code is related to leaf nodes (and unrelated to the feature in this PR). 
However, thats a good point, but it seems the second await statement uses the group nodes set during the first one, so thats something we should definitely revisit once the leaf nodes related backend issue is resolved to validate the behavior properly in UI.